src/python/pants/rules/core/cloc.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pathlib import Path
+from typing import Set
+
+from pants.base.build_environment import get_buildroot
+from pants.base.specs import Specs
+from pants.engine.console import Console
+from pants.engine.fs import (
+  Digest,
+  DirectoriesToMerge,
+  FileContent,
+  FilesContent,
+  InputFilesContent,
+  Snapshot,
+  UrlToFetch,
+)
+from pants.engine.goal import Goal
+from pants.engine.isolated_process import ExecuteProcessRequest, ExecuteProcessResult
+from pants.engine.legacy.graph import HydratedTargets, TransitiveHydratedTargets
+from pants.engine.rules import console_rule, rule
+from pants.engine.selectors import Get
+from pants.util.objects import datatype
+
+
+class DownloadedClocScript(datatype([('script_path', str), ('digest', Digest)])):
```

Comment:
Nit: users do not care how the ClocScript is retrieved, just that it's the cloc script. Consider dropping the `Dowloaded` prefix here. In the rule function name, in contrast, I think the prefix makes perfect sense.